src/replication.c

```diff
@@ -1568,14 +1568,17 @@ void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData,
 
         if (stillAlive == 0) {
             serverLog(LL_WARNING,"Diskless rdb transfer, last replica dropped, killing fork child.");
+            aeDeleteFileEvent(server.el, server.rdb_pipe_read, AE_READABLE);
             killRDBChild();
         }
-        /*  Remove the pipe read handler if at least one write handler was set. */
-        if (server.rdb_pipe_numconns_writing || stillAlive == 0) {
-            if (server.rdb_pipe_read != -1)
```

Comment:
i suppose it's ok, but just to be sure, why did we have the `server.rdb_pipe_read != -1` check and why do you think we no longer need it?

Reply:
because killRDBChild would reset it in the original code, but now i added else, which means that `killRDBChild` will definitely not be called when we enter here, so there's no need to determine if it's -1.

Reply:
ohh, that condition on rdb_pipe_read is new from the recent changes.. :+1: